arcgis-ios-sdk-samples/Geometry/Buffer/BufferViewController.swift

```diff
@@ -0,0 +1,150 @@
+        let outlineSymbol = AGSSimpleLineSymbol(style: .solid, color: .black, width: 2)
+        let fillSymbol = AGSSimpleFillSymbol(style: .solid, color: .red, outline: outlineSymbol)
+        overlay.renderer = AGSSimpleRenderer(symbol: fillSymbol)
+        overlay.opacity = 0.5
+        return overlay
+    }()
+    
+    /// The overlay for displaying the geometries created via a geodesic buffer around the tap point.
+    /// Green.
+    private let geodesicOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+        let outlineSymbol = AGSSimpleLineSymbol(style: .solid, color: .black, width: 2)
+        let fillSymbol = AGSSimpleFillSymbol(style: .solid, color: .green, outline: outlineSymbol)
+        overlay.renderer = AGSSimpleRenderer(symbol: fillSymbol)
+        overlay.opacity = 0.5
+        return overlay
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = [
+            "BufferViewController",
+            "BufferOptionsViewController"
+        ]
+        
+        // instantiate a map using a topo basemap
+        let map = AGSMap(basemap: .topographic())
+        // assign the map to the map view
+        mapView.map = map
+        
+        // add the graphics overlays in the order we want them to draw, bottom to top
+        mapView.graphicsOverlays.addObjects(from: [geodesicOverlay, planarOverlay, tapLocationsOverlay])
+        
+        // register to receive callbacks for pointer inputs in the map view
+        mapView.touchDelegate = self
+    }
+    
+    // MARK: - Navigation
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let controller = segue.destination as? BufferOptionsViewController {
+            // setup the options controller
+            controller.delegate = self
+            controller.bufferDistance = bufferDistance
+            controller.presentationController?.delegate = self
+            controller.preferredContentSize = CGSize(width: 300, height: 140)
+        }
+    }
+    
+    @IBAction func clearAllAction(_ sender: UIBarButtonItem) {
+        // remove the graphics from all the overlays, resetting the map view to its initial contents
+        for overlay in mapView.graphicsOverlays as! [AGSGraphicsOverlay] {
+            overlay.graphics.removeAllObjects()
+        }
+    }
+    
+}
+
+extension BufferViewController: AGSGeoViewTouchDelegate {
+    
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        
+        /// The buffer radius converted to meters.
+        let bufferRadiusInMeters = bufferDistance.converted(to: .meters).value
+       
+        // ensure that the buffer radius is a positive value
+        guard bufferRadiusInMeters > 0 else {
```

Comment:
Were you seeing a negative measurement? I'm not sure how that would be possible.

Reply:
It shouldn't be, but I figure this is a fine sanity check to show developers the value should always be positive.